### PR TITLE
1.1 spot ml performance improvements

### PR DESCRIPTION
**Pull request by rabarona**
## Performance improvements

After a week of code reviewing, we found few places where we could make things differently in order to improve spot-ml performance. These changes happened mainly in SpotLDAWrapper object which was collecting the list of document-word-count to save into a local file model.dat.

This pull request implements a new technic and saves model.data to HDFS first and then using `hadoop fs -copyToLocal` is saving the model.dat file locally so MPI/LDA can read it.

Among other changes, the output of SpotLDAWrapper has changed. Now it returns a data frame for the list of ip to topic distribution.

### CODE REVIEW

File: spot-ml/src/main/scala/org/apache/spot/SuspiciousConnects.scala [hunk 1]
```diff
@@ -36,7 +36,7 @@ object SuspiciousConnects {
     parser.parse(args, SuspiciousConnectsConfig()) match {
       case Some(config) =>
 
-        Logger.getLogger("org").setLevel(Level.OFF)
+        Logger.getLogger("org").setLevel(Level.WARN)
```
Comment (NathanSegerlind) [inline spot-ml/src/main/scala/org/apache/spot/SuspiciousConnects.scala:39]: how does this change the observed output of the logger?

File: spot-ml/src/main/scala/org/apache/spot/SuspiciousConnectsArgumentParser.scala [hunk 2]
```diff
@@ -95,6 +96,10 @@ object SuspiciousConnectsArgumentParser {
       action((x, c) => c.copy(hdfsScoredConnect = x)).
       text("HDFS path for results")
 
+    opt[String]("tempmodel").required().valueName("<hdfs path>").
```
Comment (NathanSegerlind) [inline spot-ml/src/main/scala/org/apache/spot/SuspiciousConnectsArgumentParser.scala:99]: going forward... we might want to think of a way to reduce the CLIs for this jar.... we are nearing the "22 field limit" that Scala has for its build in tuples and case classes

File: spot-ml/src/main/scala/org/apache/spot/dns/model/DNSSuspiciousConnectsModel.scala [hunk 3]
```diff
@@ -206,8 +207,21 @@ object DNSSuspiciousConnectsModel {
...
       config.analysis,
       config.nodes,
-      config.ldaPRGSeed)
-
+      config.ldaPRGSeed,
+      sparkContext,
+      sqlContext,
+      logger)
+
+    // Since DNS is still broadcasting ip to topic mix, we need to convert data frame to Map[String, Array[Double]]
```
Comment (NathanSegerlind) [inline spot-ml/src/main/scala/org/apache/spot/dns/model/DNSSuspiciousConnectsModel.scala:215]: going forward, we should evaluate having all pipelines use the same structure.... presumably catalyst optimzer knows when to use broadcast joins and when to do other kinds of joins

Comment (rabarona) [reply spot-ml/src/main/scala/org/apache/spot/dns/model/DNSSuspiciousConnectsModel.scala:215]: I agree

File: spot-ml/src/main/scala/org/apache/spot/SuspiciousConnectsArgumentParser.scala (hunk 2 above)
Comment (rabarona) [reply spot-ml/src/main/scala/org/apache/spot/SuspiciousConnectsArgumentParser.scala:99]: Could we take advantage of @natedogs911 's python substitute of ml_ops.sh and create a settings.txt file on the fly? That's just from the top of my head. In general I see what you are saying.

File: spot-ml/src/main/scala/org/apache/spot/SuspiciousConnects.scala (hunk 1 above)
Comment (rabarona) [reply spot-ml/src/main/scala/org/apache/spot/SuspiciousConnects.scala:39]: Based on Vartikas recommendation, we should leave `org` at least at WARN level. If there are no warnings, the output looks exactly the same.

**Review by NathanSegerlind — APPROVED** [review]

**Pull request merged**